### PR TITLE
TFSID 1470733: [OAuth] [Connector SDK] Update Connector SDK to contain updated version of OAuthConfig.xsd file for validation.

### PR DESCRIPTION
**Pull request by ethan-pyke**
…rl fields

**Things To Remember:** 

1. Be sure to merge into the right branch. The choices are:
* **`master`** if your PR runs on the current version of Tableau.
* **`dev-2021.2`**  
2. You should update at least one of the following:
* The changelog of the project you modified.
* The root `README.md` if the PR is a release of the SDK, Connector Packager, or TDVT.  
=====

TFSID 1470733: [OAuth] [Connector SDK] Update Connector SDK to contain updated version of OAuthConfig.xsd file for validation.

### CODE REVIEW

File: validation/oauth_config.xsd
Comment (rosswbrown) [inline validation/oauth_config.xsd:38]: Do we have an item to ensure the [connector-packager](https://github.com/tableau/connector-plugin-sdk/tree/master/connector-packager) sets `min-version-tableau` to 2023.1 if `defaultInstanceUrl` usage is detected?

File: validation/oauth_config.xsd
Comment (rosswbrown) [inline validation/oauth_config.xsd:28]: Ensure there is test coverage in the connector-packager XSD validation. You may be able to augment one of the existing [oauth tests](https://github.com/tableau/connector-plugin-sdk/blob/master/connector-packager/tests/test_xsd_validator.py#L124) and its [files](https://github.com/tableau/connector-plugin-sdk/blob/master/connector-packager/tests/test_resources/oauth_connector/oauth-config.xml), or make a new test, depending on the plan for the one below.

File: validation/oauth_config.xsd
Comment (ethan-pyke) [reply validation/oauth_config.xsd:38]: I think we are actually going to hold off on adding defaultInstanceUrl for now until we can do more testing on it.  Should I still do this if oauthConfigId is detected?

File: connector-packager/connector_packager/jar_jdk_packager.py
Comment (rosswbrown) [inline connector-packager/connector_packager/jar_jdk_packager.py:62]: `oauthConfigId` has existed in some form since, XSD at least, since 21.4. Normally the SDK takes the most permissive stance for compatibility. I'm unsure from the context I have what the right choice is here. 

Think about an existing connector with a min-tableau-version of 21.4 and beginning to adopt this feature. Is this a breaking change in context of those older products, or just graceful fallback to default/existing behavior.

**Review by pvanderknyff — APPROVED** [review]
LGTM once Ross's comment is addressed. Packager changes look good and well-tested.

File: connector-packager/connector_packager/jar_jdk_packager.py
Comment (pvanderknyff) [reply connector-packager/connector_packager/jar_jdk_packager.py:62]: FWIW, I don't see `oauthConfigId` mentioned in the documentation or the samples on the SDK, and up until the SDK's XSD were change in this PR connectors using it would have been rejected, so I think the odds of a partner connector having this element before 2023.1 would be very small. Still worth confirming that we're not breaking existing connectors, though.

Comment (rosswbrown) [reply connector-packager/connector_packager/jar_jdk_packager.py:62]: I was thinking about this within the context of adding functionality to a connector that already exists. I'll defer to the team on what the expected future usage is and if this is right choice.

Comment (ethan-pyke) [reply connector-packager/connector_packager/jar_jdk_packager.py]: After talking more with my team, we decided to change it to 21.4.  While we are still not sure if it will work at the moment, we decided it's better to keep it loose rather than restricting it to 23.1+.  I believe there will be some follow-up stories to ensure that earlier versions can work.

File: connector-packager/connector_packager/jar_jdk_packager.py
Comment (rosswbrown) [inline connector-packager/connector_packager/jar_jdk_packager.py:62]: This will work today, but to ensure we don't need to change it in the future, you'll want to check that min_version_tableau isn't already something larger than 2021.4 before making the set call.

**Review by rosswbrown — APPROVED** [review]

**Pull request merged**
